src/server.js

```diff
@@ -215,5 +224,76 @@ app.get(
   }
 );
 
+app.get(
+  `${PREFIX_API}/:${URL_PARAM_ADDRESS}/geocode`,
+  async (request, response) => {
+    let address = request.params[URL_PARAM_ADDRESS];
+
+    let geocodeRequest =
+      "https://maps.googleapis.com/maps/api/geocode/json?address=" +
+      address +
+      "&key=" +
+      env.API_KEY;
+
+    const geocodeResponse = await (await fetch(geocodeRequest)).json();
+
+    try {
+      response.json({
+        status: 200,
+        data: geocodeResponse.results[0].geometry.location,
+      });
+    } catch (err) {
+      console.error(err);
+      response
+        .status(500)
+        .json({ status: 500, error: { type: ERROR_GEOCODING_FAILED } });
+    }
+  }
+);
+
+/** Some Hard-coded data to display as restaurants temporarily. */
+function getRestaurants() {
+  const restaurant1 = {
+    name: "Nice Cafe",
+    address: "123 N. Street st.",
+    latLong: { lat: 37.0, lng: -122.0 },
+    rating: 5,
+    priceLevel: 2,
+    openingHours: "8:00am - 10:00pm",
+  };
+
+  const restaurant2 = {
+    name: "Bad Cafe",
+    address: "987 S. Street st.",
+    latLong: { lat: -37.0, lng: 122.0 },
+    rating: 2,
+    priceLevel: 4,
+    openingHours: "12:00am - 9:00am",
+  };
+
+  return [restaurant1, restaurant2];
+}
+
+app.get(
+  `${PREFIX_API}/:${URL_PARAM_EVENT_ID}/restaurants`,
+  async (request, response) => {
+    const restaurants = getRestaurants();
+    response.json({
+      status: 200,
+      data: Object.values(restaurants),
+    });
+  }
+);
+
 const port = 8080;
-app.listen(port, () => console.log(`Server listening on port: ${port}`));
+const server = app.listen(port, () =>
+  console.log(`Server listening on port: ${port}`)
+);
+
+const io = require("socket.io")(server);
+
+io.on("connection", (socket) => {
```

Comment:
this could use a comment explaining what this is intended to do at a high levevl

Reply:
I added some comments on the server and in searchPage.js to explain a little more what is going on. Let me know if there is too much or too little explanation here.